src/main.cpp

```diff
@@ -875,6 +905,18 @@ int main(int argc, char **argv) {
                 codegen_link(g, out_file);
                 if (timing_info)
                     codegen_print_timing_report(g, stdout);
+
+                if (cmd == CmdRun) {
+                    ZigList<const char*> args = {0};
+                    for (int i = runtime_args_start; i < argc; ++i) {
+                        args.append(argv[i]);
+                    }
+
+                    Termination term;
+                    os_spawn_process(buf_ptr(run_exec_path), args, &term);
```

Comment:
This is probably a good first step, and I'm happy to merge with it like this, but 2 questions:

 * Do you think we should use execve? Completely replace the current process with the new one?
 * If we did this, what about Windows?

I'm thinking about the case where the process we're running gets SIGABRT. With this code, zig would call exit 1. Still exiting with an error, but a different way. Maybe it's fine though. I think if you usd gdb with follow-fork-child, it would still catch the abort.

Reply:
Good point. I'll need to look more into the small details to see if there is a way we could manage that use case. Will leave for now.